### DISCUSSION TITLE
How to create a named HttpSecurityPolicy that can be used with quarkus.http.auth

### DISCUSSION THREAD

**Discussion by fediazgon** [quarkusio/quarkus#27035] [Q&A]
At the moment, I have the following `HttpSecurityPolicy` which is invoked on every request:

```java
import io.quarkus.security.identity.SecurityIdentity;
import io.quarkus.vertx.http.runtime.security.HttpSecurityPolicy;
import io.smallrye.mutiny.Uni;
import io.vertx.ext.web.RoutingContext;
import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class SecurityHandler implements HttpSecurityPolicy {

  @Override
  public Uni<CheckResult> checkPermission(
      final RoutingContext request,
      final Uni<SecurityIdentity> identity,
      final AuthorizationRequestContext requestContext) {
    return ...
  }
}
```

According to the [HttpSecurityPolicy documentation](http://javadox.com/io.quarkus/quarkus-vertx-http/1.1.0.Final/io/quarkus/vertx/http/runtime/security/HttpSecurityPolicy.html), if I created a named policy, it can then be referenced in the `application.properties` path matching rules, which allows this policy to be applied to specific requests. Same thing is explained in the [quarkus.http.auth.permission.-permissions-.policy](https://quarkus.io/guides/all-config#quarkus-vertx-http_quarkus.http.auth.permission.-permissions-.policy) documentation.

So then, is it possible to create a named `mypolicy` policy that I can later reference like:

```
quarkus.http.auth.permission.mypermission.policy=mypolicy
```

I've tried with `@Named` but is not working. I cannot use `HttpSecurityPolicyBuildItem` because I'm not developing an extension. It seems that Quarkus read all available policies in [HttpSecurityProcessor](https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpSecurityProcessor.java#L134), but I have no idea how I can add my policy to that `policyMap`.

[a comment by an automated account is not shown]

**Comment by akinevz2** [comment]
Override the `name()` method
